theforeman/foreman
# Fixes #30771 - Global registration - New Workflow

Updated WF for registering host via Global Registration template and Host Registration Template without SmartProxy.
For more information & background, see this [RFC comment](https://community.theforeman.org/t/rfc-simple-automatic-host-registration-wf/19588/33) and following discussion.

**Summary**
* UC 1 - Registering host with Vanilla Foreman
* UC 2 - Registering host on Foreman with Katello, (For this you need https://github.com/Katello/katello/pull/8965)
* New endpoint `POST /register` for finding/creating `@host` and rendering Host registration template
* _GRT_ improvements (Better error handling, added `operatingsystem_id` parameter ...
* _Host registration template_ - Added installation of insights and REX setup
* New common parameter _host_registration_insights_
* Some refactoring & test coverage ...

![7b75f85e75f875690a910566e7d8654778fa8c51](https://user-images.githubusercontent.com/59385976/94102744-2245ac00-fe33-11ea-9e29-c00019f5ec69.png)

**Not covered in this PR**
* Scenarios with Proxy
* Secure connection
* Validation of parameters
* Any other feature from [Redmine](https://projects.theforeman.org/issues/30440)

**How to test**
* Create OS and assign it a registration template
* On your host run command:
```
curl -X GET "foreman.example.com/register?organization_id=1&location_id=2&operatingsystem_id=OS_ID" --user admin:changeme | bash
```

---
PR is part of **Simple & automatic host registration WF**, see:
* [RFC](https://community.theforeman.org/t/rfc-simple-automatic-host-registration-wf/19588)
* [Redmine](https://projects.theforeman.org/issues/30440)
* [Redmine (this PR)](https://projects.theforeman.org/issues/30771)
